### PR TITLE
Quick Start V2: update checklist

### PR DESCRIPTION
**Pull request by danielebogo**
Refs. #10860 

This PR updates the current quick start checklist. It uses 2 sections to divide the completed tasks from the todo tasks.

qs-section

Because this update is behind feature flag, I created a new data flow _QuickStartChecklistManager_ to handle the table view datasource and delegate.

@SylvesterWilmott this is the final result:

simulator screen shot - iphone x - 2019-01-24 at 13 52 16

simulator screen shot - iphone x - 2019-01-24 at 13 52 18

**To test:**
- Create a new site and turn on quick start (or put return true in shouldShowQuickStartChecklist())
- Select _Customize Your Site_ to complete at least one task
- You should see the todo list separated from the completed list
- Tap on the _Complete_ header to expand/hide the section

### CODE REVIEW

**Comment by SylvesterWilmott** [comment]
This looks fantastic @danielebogo thanks!

There seems to be an issue with the separators:
- The separators don't align with the text
- The separator on the first item in the completed list is full width

**Comment by danielebogo** [comment]
@SylvesterWilmott I'm fixing it!

**Comment by danielebogo** [comment]
@SylvesterWilmott here:

simulator screen shot - iphone x - 2019-01-24 at 17 41 50

File: WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        self.didTapHeader = didTapHeader
+        super.init()
+        reloadData()
+    }
+
+    func reloadData() {
+        let completed = (blog.completedQuickStartTours ?? []).map { $0.tourID }
+        let skipped = (blog.skippedQuickStartTours ?? []).map { $0.tourID }
+
+        completedToursKeys = Set(completed).union(Set(skipped))
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift]: I think there is a logic problem here. `blog.skippedQuickStartTours` are tours that have been skipped from the notices. This isn't supposed to mark them as complete. This is different than the `Skip` button(s) in the UI, which _do_ mark them as complete. That's a difference between what "skip" means in the code vs. the UI. 

In the new UI, the sideways UI will mark them as complete, as for all our purposes they will be. Let me know if that doesn't make sense.

File: WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        } else {
+            tableView.deleteRows(at: indexPaths, with: .fade)
+        }
+
+        didTapHeader(collapsing)
+    }
+}
+
+private enum Sections: Int, CaseIterable {
+    static let footerHeight = CGFloat(20.0)
```
[the remark this answers is not in the document]
Comment (danielebogo) [reply WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift]: This is actually the footer view height of the first section 😊

**Comment by SylvesterWilmott** [comment]
@danielebogo let's just change the stroke colour to `greyLighten20`

File: WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift (hunk 1 above)
Comment (SylvesterWilmott) [reply WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift]: Could there be a benefit of making a distinction between skipped and completed for any future iterations we might make?

**Comment by SylvesterWilmott** [comment]
Thanks @danielebogo !

File: WordPress/Classes/ViewRelated/Blog/QuickStartChecklistViewController.swift [hunk 3]
```diff
@@ -1,21 +1,45 @@
...
+    }
+}
+
 class QuickStartChecklistViewController: UITableViewController {
     private var dataSource: QuickStartChecklistDataSource? {
         didSet {
             self.tableView?.dataSource = dataSource
         }
     }
+    private var dataManager: QuickStartChecklistManager? {
```
[the remark this answers is not in the document]
Comment (danielebogo) [reply WordPress/Classes/ViewRelated/Blog/QuickStartChecklistViewController.swift]: Ok

**Comment by danielebogo** [comment]
@nheagy I fixed everything and duplicate the `QuickStartChecklistViewController`. I also changed the way this new vc is instantiated. It uses an enum (easy to use by objc classes) rather than injecting the configuration structs.

**Comment by danielebogo** [comment]
Thanks @nheagy @SylvesterWilmott

**Pull request merged**